src/Pos/Communication/Limits.hs

```diff
@@ -209,9 +228,9 @@ instance MessageLimited (DataMsg GtMsgContents) where
     getMsgLenLimit _ =
         return $ each %~ fmap DataMsg $
             ( mcCommitmentMsgLenLimit
-            , noLimit
+            , NoLimit
```

Comment:
> Do you mean that it contains AsBinary Secret?

:O I didn't know there is `AsBinary`.

> he would most probably forget to change limit if it's just a magic constant

I expected it to be caught by tests.

> which is unpleasant for complex types

Well, for complex types we probably don't want to use `NoLimit`. And for simple types I see this: http://i.imgur.com/staJTar.png

> Then why a "value such that we can easily parse so many bytes and at the same time it can fit all fixed-size data" was bad?

http://i.imgur.com/isfqmMm.png
Why did I say it was bad? 😉 
I can't say now which solution I find the best, I want to sleep so will get back to it tomorrow.